error_translator.go

```diff
@@ -0,0 +1,35 @@
+package postgres
+
+import (
+	"encoding/json"
+	"gorm.io/gorm"
+)
+
+var errCodes = map[string]string{
+	"uniqueConstraint": "23505",
+}
+
+type ErrMessage struct {
+	Code     string `json:"Code"`
+	Severity string `json:"Severity"`
+	Message  string `json:"Message"`
+}
+
+func (dialector Dialector) Translate(err error) error {
+	parsedErr, marshalErr := json.Marshal(err)
```

Comment:
Maybe better to assert the error type for performance? e.g:

`pgErr, ok := err.(*pgconn.PgError)`

And use JSON marshal/unmarshal only when failing to assert

Reply:
Yes it is a good idea, I will update the PR

Reply:
yes, but it might be broken if using a different pg driver... but anyway, both are fine for me.

Reply:
```go
		var p *pgconn.PgError
		if errors.As(err, &p) && p.Code == "23505" {
			if strings.Contains(p.Detail, "email") {
				return nil, errno.DBError(errno.ErrEmailExist, "注册商户失败")
			}
			if strings.Contains(p.Detail, "phone") {
				return nil, errno.DBError(errno.ErrMerchantPhoneExist, "注册商户失败")
			}
		}
```
How should I tell which key is conflicting in the new version ?

Reply:
In fact I think this has been breaking API, The original code cannot be reached after updating the new version,Resulting in users not knowing whether the mailbox is duplicated or the phone number is duplicated,So I use v1.4.8

Reply:
We can enhance the error to have more detail, @jinzhu what do you think?

Reply:
+1 @Naist4869. Previously, I could parse pgConn.pgError for the constraint name and handle each constraint violation differently.

Reply:
Yes, agree this is a breaking API, the good news we still haven't released a new version for GORM, so these changes won't affected users unless they are using the latest master.

Maybe we need to add a configuration `TranslateError` to GORM Config, https://github.com/go-gorm/gorm/blob/master/gorm.go#L20, by default we don't translate the error.

@saeidee do you feel good to make the change?

Reply:
Yes, I can implement the changes

Reply:
Here is the PR https://github.com/go-gorm/gorm/pull/6178